kube-core/src/metadata.rs

```diff
@@ -12,3 +14,50 @@ pub struct TypeMeta {
     /// The name of the API
     pub kind: String,
 }
+
+/// Generic representation of any object with ObjectMeta. Allows clients to get
+/// access to a particular ObjectMeta schema without knowing the details of the
+/// version
+#[derive(Deserialize, Serialize, Clone, Default, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PartialObjectMeta {
+    /// The type fields, not always present
+    #[serde(flatten, default)]
+    pub types: Option<TypeMeta>,
+    /// Object metadata
+    #[serde(default)]
+    pub metadata: ObjectMeta,
+}
+
+/// Represents a list of objects and contains only the ObjectMeta
+#[derive(Deserialize, Serialize, Clone, Default, Debug)]
+pub struct PartialObjectMetaList {
+    /// The type fields
+    #[serde(flatten, default)]
+    pub types: Option<TypeMeta>,
```

Comment:
AFAIKT this is the inclusion of this type (that we do not use) is the only difference between `PartialObjectMetaList<PartialObjectMeta>` and `ObjectList`. Given there are no other bounds constraints on `ObjectList` i'm leaning towards just using that unless there's more info in the partial that __could__ be useful.

That would avoid the (largely pointless-feeling) converter between them.

Reply:
While upstream does have a specific List type for this as well, it follows [their regular list type conventions](https://github.com/kubernetes/api/blob/0478a3e95231398d8b380dc2a1905972be8ae1d5/apps/v1/types.go#L551-L560), so we should re-use `ObjectList<T>`. We have used `ObjectList` as the generic representation of that struct for ages and it continues to fit here with no extra fields.

Reply:
Cool! I made the change. Didn't really do an impl for `From<ObjectList<PartialObjectMeta>> for ObjectList<PartialObjectMeta>` since I'm not sure we need it? Users can do a map or access the fields directly. If it makes sense to have as a util when we do the watcher I might add it back. Can also do that as part of this change if you think it's a good idea.

Reply:
> `From<ObjectList<PartialObjectMeta>> for ObjectList<PartialObjectMeta`

uh, do you mean:
`From<ObjectList<PartialObjectMeta>> for ObjectList<ObjectMeta>` ?

if so, then nah, this is fine.

Reply:
Yeah, oops meant `for ObjectList<ObjectMeta>`. All the objects and the metas got to me. But cool, makes sense 👍🏻